chia/full_node/full_node.py

```diff
@@ -699,7 +701,11 @@ async def short_sync_backtrack(
                         f"Failed to fetch block {curr_height} from {peer.get_peer_logging()}, wrong type {type(curr)}"
                     )
                 blocks.append(curr.block)
-                if self.blockchain.contains_block(curr.block.prev_header_hash) or curr_height == 0:
+                if curr_height == 0:
+                    found_fork_point = True
+                    break
+                hash_at_height = self.blockchain.height_to_hash(curr.block.height - 1)
+                if hash_at_height is not None and hash_at_height == curr.block.prev_header_hash:
```

Comment:
as long as `curr.block.prev_header_hash` is never `None`, I imagine the first part of this condition is unnecessary (`hash_at_height is not None`). if it is `None` it wouldn't compare equal to a `uint32`. Not a big deal though

Reply:
since this is backtrack sync this height may be above our peak so None is a possibility, we are trying to find the first height we have with an equal header hash

Reply:
`None == curr.block.prev_header_hash` would always evaluate to `False`, so the first factor, before `and` is technically not necessary. But maybe this is a bit clearer, the reader doesn't need to know that `curr.block.prev_header_hash` is never `None` for instance.

Reply:
iir the linter doesn't like the comparison between an optional and a uint32